index/index.go

```diff
@@ -193,59 +121,98 @@ func (index *Index) addFile(subPath string) error {
 	return nil
 }
 
-func newIndexer(path string, fileIndex int, index *Index) indexer {
-	indexer := indexer{
-		index:     index,
-		fileIndex: fileIndex,
-		path:      path,
+func (index *Index) buildIndexOverFile(fileId int, f indexFile) error {
+	entry := make([]byte, 12)
+
+	for {
+		offset, err := f.file.Seek(0, 1)
+		if err != nil {
+			return err
+		}
+
+		if !f.reader.ScanKey() {
+			break
+		}
+
+		serializeIndexEntry(entry, fileId, offset)
+		index.ldb.Put(f.reader.Key(), entry, nil)
+		index.count++
+	}
+
+	if err := f.reader.Err(); err != nil {
+		return err
 	}
 
-	return indexer
+	log.Println("Finished indexing", f.file.Name())
```

Comment:
Should this log go in the calling function? You log `Indexing...` in the caller. If you move it out you can `return f.reader.Err()`.